### PR TITLE
Add tcp and udp moudule Integration tests and BBE

### PR DESCRIPTION
**Pull request by MohamedSabthar**

### CODE REVIEW

File: examples/tcp-listener/tcp_listener.bal [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    remote function onClose() returns tcp:Error? {
+         io:println("Client left: ", self.caller.remotePort);
+    }
+}
+
+function getString(byte[] content) returns @tainted string|io:Error {
+    io:ReadableByteChannel byteChannel = check io:createReadableChannel(content);
+    io:ReadableCharacterChannel characterChannel = new io:ReadableCharacterChannel(byteChannel, "UTF-8");
+    return check characterChannel.read(content.length());
+}
```
Comment (Maninda) [inline examples/tcp-listener/tcp_listener.bal:44]:
```suggestion
}

```

File: stdlib-integration-tests/udp/tests/udp_tests.bal [hunk 2]
```diff
@@ -110,3 +110,20 @@ isolated function testConnectClientReadTimeOut() {
...
+        if (str is string) {
+            returnStr = <@untainted>str;
+            io:println("Response is :", returnStr);
+        } else {
+            test:assertFail(msg = str.message());
+        }
+    } else {
+        test:assertFail(msg = "Failed to receive the data");
+    }
+    return returnStr;
```
Comment (Maninda) [inline stdlib-integration-tests/udp/tests/udp_tests.bal:128]:
```suggestion
    return returnStr;
    
```

File: stdlib-integration-tests/udp/tests/udp_tests.bal [hunk 3]
```diff
@@ -110,3 +110,20 @@ isolated function testConnectClientReadTimeOut() {
...
+}
```
Comment (Maninda) [inline stdlib-integration-tests/udp/tests/udp_tests.bal:129]:
```suggestion
}

```

File: examples/tcp-listener/tcp_listener.bal (hunk 1 above)
Comment (MohamedSabthar) [reply examples/tcp-listener/tcp_listener.bal]: added new line

File: stdlib-integration-tests/udp/tests/udp_tests.bal (hunk 3 above)
Comment (MohamedSabthar) [reply stdlib-integration-tests/udp/tests/udp_tests.bal]: added new line

File: examples/tcp-client/tcp_client.bal [hunk 4]
```diff
@@ -0,0 +1,33 @@
+// This is the client implementation for the TCP socket.
```
Comment (praneesha) [inline examples/tcp-client/tcp_client.bal:1]:
```suggestion
// This is the client implementation of the TCP socket.
```

File: examples/tcp-listener/tcp_listener.bal [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+    // This remote method is invoked once the content is received from the client.
+    remote function onBytes(readonly & byte[] data) returns tcp:Error? {
+        io:println("Echo: ", getString(data));
+        // Echo back the data to the same client which the data received from.
+        check self.caller->writeBytes(data);
+    }
+
+    // This remote method is invoked in an error situation
+    // if it happens during the `onConnect` and `onBytes`.
```
Comment (praneesha) [inline examples/tcp-listener/tcp_listener.bal:29]:
```suggestion
    // if it happens during the `onConnect` and `onBytes` methods.
```

File: examples/tcp-listener/tcp_listener.description [hunk 6]
```diff
@@ -1,5 +1,4 @@
 // The TCP Listener is used to expose TCP service over the TCP protocol.
```
Comment (praneesha) [inline examples/tcp-listener/tcp_listener.description:1]:
```suggestion
// The TCP Listener is used to expose a TCP service over the TCP protocol.
```

File: examples/udp-connect-client/udp_connect_client.bal [hunk 7]
```diff
@@ -0,0 +1,53 @@
+// This is the connection oriented client implementation for the UDP socket.
```
Comment (praneesha) [inline examples/udp-connect-client/udp_connect_client.bal:1]:
```suggestion
// This is the connection-oriented client implementation of the UDP socket.
```

File: examples/udp-listener/udp_listener.bal [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+// Bind the service to the port.
+// Optionally, you can provide the `remoteHost` and `remotePort` to
+// configure the listener as a connected listener, which only
+// reads and writes to the configured remote host.
+// udp:Listener(48829, remoteHost = "www.remote-clinet.com", remotePort = 8080)
+service on new udp:Listener(48829) {
+
+    // This remote method is invoked once the content is received from the client.
+    // You may replace the onBytes method with onDatagram which reads the data as
```
Comment (praneesha) [inline examples/udp-listener/udp_listener.bal:14]:
```suggestion
    // You may replace the `onBytes` method with `onDatagram`, which reads the data as
```

File: examples/tcp-client/tcp_client.bal (hunk 4 above)
[the remark this answers is not in the document]
Comment (MohamedSabthar) [reply examples/tcp-client/tcp_client.bal:1]: removed

File: examples/tcp-client/tcp_client.bal [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    string stringData = check characterChannel.read(receivedData.length());
+    io:println("Recived: ", stringData);
+
+    // Close the connection between the server and the client.
+    var closeResult = socketClient->close();
+    if (closeResult is error) {
+        io:println("An error occurred while closing the socket ", closeResult);
+    }
+}
+
```
[the remark this answers is not in the document]
Comment (MohamedSabthar) [reply examples/tcp-client/tcp_client.bal:33]: removed

File: examples/tcp-client/tcp_client.out [hunk 10]
```diff
@@ -1,7 +1,6 @@
 # To run the client, navigate to the directory that contains the
 # `.bal` file and use the `ballerina run` command below.
```
[the remark this answers is not in the document]
Comment (MohamedSabthar) [reply examples/tcp-client/tcp_client.out:2]: fixed in udp,tcp examples

File: examples/tcp-listener/tcp_listener.bal [hunk 11]
```diff
@@ -0,0 +1,44 @@
+// This is the server implementation for the TCP socket.
```
[the remark this answers is not in the document]
Comment (MohamedSabthar) [reply examples/tcp-listener/tcp_listener.bal:1]: removed

File: examples/udp-client/udp_client.bal [hunk 12]
```diff
@@ -2,38 +2,36 @@
...
+        remoteHost: "localhost",
+        remotePort : 48829,
         data : msg.toBytes()
     };
 
-    // Send data to remote host.
-    // The parameter is a Datagram record which contains the remote address
-    // and the data to be.
-    var sendResult = socketClient->send(datagram);
+    // Send the data to remote host.
```
[the remark this answers is not in the document]
Comment (MohamedSabthar) [reply examples/udp-client/udp_client.bal:21]: updated

File: examples/tcp-client/tcp_client.bal (hunk 4 above)
Comment (MohamedSabthar) [reply examples/tcp-client/tcp_client.bal:1]: updated

File: examples/tcp-listener/tcp_listener.bal (hunk 5 above)
Comment (MohamedSabthar) [reply examples/tcp-listener/tcp_listener.bal:29]: updated

File: examples/udp-connect-client/udp_connect_client.bal (hunk 7 above)
Comment (MohamedSabthar) [reply examples/udp-connect-client/udp_connect_client.bal:1]: updated

File: examples/tcp-listener/tcp_listener.description (hunk 6 above)
Comment (MohamedSabthar) [reply examples/tcp-listener/tcp_listener.description:1]: updated

File: examples/udp-listener/udp_listener.bal (hunk 8 above)
Comment (MohamedSabthar) [reply examples/udp-listener/udp_listener.bal:14]: updated

File: examples/tcp-listener/tcp_listener.out [hunk 13]
```diff
@@ -0,0 +1,8 @@
+# To start the service, navigate to the directory that contains the
+# `.bal` file and use the `ballerina run` command below.
+bal run tcp_listener.bal
+
+# The tcp listener starts listening to the port 61598 for incoming client requests.
+Client connected to echoServer: 48735
+Echo: Hello Ballerina Echo from client
+Client left: 48735
```
[the remark this answers is not in the document]
Comment (MohamedSabthar) [reply examples/tcp-listener/tcp_listener.out:8]: new line added

**Comment by ThisaruGuruge** [comment]
Shall we resolve the conflicts, so that we can merge this?
@MohamedSabthar @Maninda @praneesha

**Comment by lafernando** [comment]
Is this PR #1140 having the same changes as here? Please check #1140 for related comments.

**Review by lafernando — CHANGES_REQUESTED** [review]
Please check and do the same changes as mentioned in the comments in #1140.

**Comment by MohamedSabthar** [comment]
> Please check and do the same changes as mentioned in the comments in #1140.

changes made according to the review.

**Comment by lafernando** [comment]
@MohamedSabthar There are some more comments there in #1140 that you haven't replied to, please check them as well.

**Comment by MohamedSabthar** [comment]
> @MohamedSabthar There are some more comments there in #1140 that you haven't replied to, please check them as well.

made the changes and replied now.

**Review by praneesha — APPROVED** [review]

**Comment by MohamedSabthar** [comment]
Duplicate of #https://github.com/ballerina-platform/ballerina-distribution/pull/1140

**Comment by Maninda** [comment]
Mistakenly opened. Will close it.

**Pull request closed without merging**